src/framework/execution/__tests__/logger.test.ts

```diff
@@ -310,3 +317,89 @@ describe('step event publishing', () => {
     });
   });
 });
+
+describe('validation failure logging', () => {
+  test('publishes message to synchronizer and writes error log', async () => {
+    const logger = createIntegrationLogger({ name, invocationConfig });
+    const context: SynchronizationJobContext = {
+      logger,
+      job: { id: 'test-job-id' } as SynchronizationJob,
+      apiClient: createApiClient({
+        apiBaseUrl: 'https://api.us.jupiterone.io',
+        account: 'mocheronis',
+      }),
+    };
+
+    logger.registerSynchronizationJobContext(context);
+
+    const errorSpy = jest.spyOn(logger, 'error');
+    const postSpy = jest
+      .spyOn(context.apiClient, 'post')
+      .mockImplementation(noop as any);
+
+    const error = new IntegrationValidationError('Bad Mochi');
+    logger.validationFailure(error);
+    await logger.flush();
+
+    const expectedDescriptionRegex = new RegExp(
+      `Error occurred while validating integration configuration. \\(errorCode=${error.code}, errorId=(.*), reason=Bad Mochi\\)$`,
+    );
+
+    expect(errorSpy).toHaveBeenCalledTimes(1);
+    expect(errorSpy).toHaveBeenCalledWith(
+      { errorId: expect.any(String), err: error },
+      expect.stringMatching(expectedDescriptionRegex),
+    );
+
+    const expectedEventsUrl =
+      '/persister/synchronization/jobs/test-job-id/events';
+
+    expect(postSpy).toHaveBeenCalledTimes(1);
+    expect(postSpy).toHaveBeenNthCalledWith(1, expectedEventsUrl, {
+      events: [
+        {
+          name: 'validation-failure',
+          description: expect.stringMatching(expectedDescriptionRegex),
+        },
+      ],
+    });
+  });
+});
+
+describe('createErrorEventDescription', () => {
+  test('adds error message to "reason" if expose option on error is set to true', () => {
+    const errorWithExposeTrue = new Error('soba');
+    (errorWithExposeTrue as any).expose = true;
+
+    const { description, errorId } = createErrorEventDescription(
+      errorWithExposeTrue,
+      'testing',
+    );
+    expect(description).toEqual(
+      `testing (errorCode=${UNEXPECTED_ERROR_CODE}, errorId=${errorId}, reason=soba)`,
+    );
+  });
+
+  test('does not add reason if expose is not set', () => {
```

Comment:
Why tho?  Because the stack / message could be really large?

Reply:
I'm not really worried about the message length too much and we don't display the stack in published logs.

I thought about allowing all error messages to be shown, but settled on only allowing for certain errors to be exposed to users. I didn't want some silly javascript/typescript error to end up getting surfaced. Surfacing errors like that make us look pretty bad and is often not useful for end users.

Reply:
I think this is a good question. Perhaps looking at older integrations we could find cases where we don't expose the message, but I wonder if we could always just expose it. I wouldn't expect the `message` to be multi-line.

Reply:
I think that we want to catch all errors and for those which are unexpected, the message would be something like https://bitbucket.org/lifeomic/jupiter-managed-integration-sdk/src/c53e0f7e0403b636adcf004798b0a49c8b9d8753/src/integration/types/errors.ts#lines-41. That is, I agree, we want to control what is shown for sure, but it seems there is always a message we can show.

Reply:
I suppose we can just replace the logged message with a canned one if we find that the error throw is not an instance of `IntegrationError`

Reply:
> I'm not really worried about the message length too much and we don't display the stack in published logs.
> 
> I thought about allowing all error messages to be shown, but settled on only allowing for certain errors to be exposed to users. I didn't want some silly javascript/typescript error to end up getting surfaced. Surfacing errors like that make us look pretty bad and is often not useful for end users.

Okay yea.  We'll still receive the error in logs and something like `Type error: cannot read property "blahablahblah" of null` looks bad.